provisioning/service/src/DeviceRegistrationsClient.cs

```diff
@@ -0,0 +1,151 @@
+
+using System;
+using System.Globalization;
+using System.Net.Http;
+using System.Net;
+using Newtonsoft.Json;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace Microsoft.Azure.Devices.Provisioning.Service
+{
+    /// <summary>
+    /// Subclient of <see cref="ProvisioningServiceClient"/> that handles all device registration state operations including
+    /// getting a device registration state, deleting a device registration state, and querying device registration states.
+    /// </summary>
+    public class DeviceRegistrationStatesClient
+    {
+        private const string ServiceName = "registrations";
+        private const string DeviceRegistrationStatusUriFormat = "{0}/{1}?{2}";
+
+        private readonly IContractApiHttp _contractApiHttp;
+        private readonly ServiceConnectionString _serviceConnectionString;
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected DeviceRegistrationStatesClient()
+        {
+        }
+
+        internal DeviceRegistrationStatesClient(ServiceConnectionString serviceConnectionString, IContractApiHttp contractApiHttp)
+        {
+            _serviceConnectionString = serviceConnectionString;
+            _contractApiHttp = contractApiHttp;
+        }
+
+        /// <summary>
+        /// Retrieve the registration status information.
+        /// </summary>
+        /// <remarks>
+        /// This method will return the DeviceRegistrationState for the provided id. It will retrieve
+        /// the correspondent DeviceRegistrationState from the Device Provisioning Service, and return it in the
+        /// <see cref="DeviceRegistrationState"/> object.
+        /// </remarks>
+        /// <param name="id">The string that identifies the DeviceRegistrationState. It cannot be null or empty.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The <see cref="DeviceRegistrationState"/> with the content of the DeviceRegistrationState in the Provisioning Device Service.</returns>
+        /// <exception cref="ArgumentNullException">If the provided <paramref name="id"/> is null.</exception>
+        /// <exception cref="ArgumentException">If the provided <paramref name="id"/> is empty or white space.</exception>
+        /// <exception cref="DeviceProvisioningServiceException">
+        /// If the service was not able to retrieve the registration state for the provided <paramref name="id"/>.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided <paramref name="cancellationToken"/> has requested cancellation.</exception>
+        public async Task<DeviceRegistrationState> GetAsync(string id, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrWhiteSpace(id, nameof(id));
+
```

Comment:
```suggestion
cancellationToken.ThrowIfCancellationRequested();
```

Reply:
Presumably, the other method he's calling just after this checks for the cancellation token. This guy's only responsibility is to check his specific parameters.